src/block-data-providers/bitcoin-core/provider.ts

```diff
@@ -122,6 +124,7 @@ export class BitcoinCoreProvider
                 }
 
                 state.indexedBlockHeight = height;
+                this.eventEmitter.emit('blockIndexed', height);
```

Comment:
`'blockIndexed'` should be a const defined in `events.ts` in common directory.
We should call it `INDEXED_BLOCK_EVENT`

Reply:
This should be emitted after the state is set.